File: _posts/arrays/best_time_to_buy_and_sell_stock_2/go/solution_min_max.go
package main

func maxProfitMinMax(prices []int) int {
	res := 0
	min := prices[0]
	max := prices[0]

	for i := 1; i < len(prices); i++ {
		if prices[i] > max {
			max = prices[i]
		} else {
			res += max - min
			min = prices[i]
			max = prices[i]
		}
	}

	res += max - min

	return res
}

File: _posts/arrays/best_time_to_buy_and_sell_stock_2/ts/solution_min_max.ts
export const maxProfitMinMax = (prices: number[]): number => {
    let res = 0
    let min = prices[0]
    let max = prices[0]

    for (let i = 1; i < prices.length; i++) {
        if (prices[i] > max) {
            max = prices[i]
        } else {
            res += max - min
            min = prices[i]
            max = prices[i]
        }
    }

    res += max - min

    return res
}
